padding/padding_test.go

```diff
@@ -0,0 +1,82 @@
+	//We can add change the size of the message by doubling this part before adding the contest of testmessage.txt, which is a bunch of text from random webpages, and project gutenberg books, that has been copy and pasted somewhat randomly to increase the size.
+	pt = append(pt, pt...)
+	pt = append(pt, pt...)
+	//Add to  our pt from a file to make it easier to test if it works for large files.
+	dat, err := ioutil.ReadFile("testmessage.txt")
+	if err != nil {
+		fmt.Println(err)
+
+	}
+	pt = append(pt, dat...)
+	ad := []byte("1234567890123456123123123123123123")
+	//nonce by default needs to be 12 bytes
+	nonce := []byte("123456781238")
+
+	aes, err := aes.NewCipher(key)
+	if err != nil {
+		fmt.Println(err)
+	}
+	aesgcm, err := cipher.NewGCMWithNonceSize(aes, len(nonce))
+	if err != nil {
+		fmt.Println(err)
+	}
+	//aesgcm adds 16 bytes to the end of the encrpyted message. These 16 bytes are the authentication tag.
+	//And an additional 8 bytes for the overhead of how we are padding it.
+	x := GetLeakBits(len(pt), 16+8)
+	y := GetMsgBits(len(pt), 16+8)
+	z := GetZeroBits(len(pt), 16+8)
+	p := GetPaddingLen(len(pt), 16+8)
+	fmt.Println("Msg w/ static overhead: ", len(pt)+16+8, " Bits to store len: ",
+		y, " Leak Bits: ", x, " ZeroBits: ", z)
+
+	fmt.Println("Padding needed: ", p)
+	fmt.Printf("Increase in message size caused by the padding (padding+len+oh)/len: %.4f\n", ((float64(p+24) + float64(len(pt))) / float64(len(pt))))
+
+	pt1 := pt
+	//PadGeneric takes in a pt, and the overhead that encryption scheme will add. It returns a p' with the format [Amount of padding(8 bytes)][original pt][padding]
```

Comment:
I think it's more readable if you keep the original in pt (you could call it 'message'), then work with another variable down. Else it can be difficult to see the difference between 'pt' after and before here.